ldap/servers/slapd/daemon.c

```diff
@@ -1554,43 +1554,54 @@ setup_pr_read_pds(Connection_Table *ct, int listnum)
         if (c->c_state == CONN_STATE_FREE) {
             connection_table_move_connection_out_of_active_list(ct, c);
         } else {
-            /* we try to acquire the connection mutex, if it is already
+            /* Check for a timeout for PAGED RESULTS */
+            if (pagedresults_is_timedout_nolock(c)) {
+                /*
+                 * There could be a race condition so lets try again with the
+                 * right lock
+                 */
+                pthread_mutex_t *mutex = pageresult_lock_get_addr(c);
```

Comment:
Instead of 'mutex' you may name it something like 'pr_mutex', to make it clear it is protecting page result stuff.